packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -566,11 +566,19 @@ export class CfnParser {
       case 'Fn::FindInMap': {
         const value = this.parseValue(object[key]);
         // the first argument to FindInMap is the mapping name
-        const mapping = this.finder.findMapping(value[0]);
-        if (!mapping) {
-          throw new Error(`Mapping used in FindInMap expression with name '${value[0]}' was not found in the template`);
+        let mappingName: string;
+        if (Token.isUnresolved(value[0])) {
+          // the first argument can be a dynamic expression like Ref: Param;
+          // if it is, we can't find the mapping in advance
+          mappingName = value[0];
+        } else {
+          const mapping = this.finder.findMapping(value[0]);
+          if (!mapping) {
```

Comment:
This error seems to seems at odds with allowing unresolved values. Is this error preventing something from breaking the road? If so, will it break if this value is unresolved?

Reply:
So, this error is to prevent situations where the argument to `Fn::FindInMap` is a static string (so something like `{ 'Fn::FinInMap': ['MappingName', 'Lvl1', 'Lvl2'] }`), but a Mapping with that name could not be found in the template (so, continuing the above example, `MappingName` would not be in the template).

In case the value is dynamic, like in the [test for this PR](https://github.com/aws/aws-cdk/pull/13428/files#diff-11b3bf6202e1ca85d83937e3c482971aeaa4eacb65513e956f7b3f6b6700ead7), we can't prevent this mistake from happening. We also can't handle things like re-naming the Mapping from CDK code correctly with dynamic mappings. But I think that's fine - we do the validation when we can, and when we can't, we just trust the template author they modeled things correctly.